### PR TITLE
Btb 45 configurator dialog

### PR DESCRIPTION
**Pull request by NSakuta**
# Issue
https://amiconsult.atlassian.net/browse/BTB-XX

## Changes

- 
- 

## Checklist

* [x] I tested the functionality on my local environment.
* [x] I made sure my code is easy to read and understand.
* [x] I provided sufficient automated tests for the code.
* [x] There a no conflicts from my-branch..main.

### CODE REVIEW

File: src/app/configurator/configurator.component.html [hunk 1]
```diff
@@ -1 +1,76 @@
...
+              </mat-form-field>
+            </div>
+
+
+            <div class="support" *ngIf="configurator !== 'free' && configurator !== 'trial'">
+              <h3 class="txt-support">Support Service: </h3>
+              <mat-checkbox formControlName="support">
+                Support
+              </mat-checkbox>
+            </div>
```
Comment (knicken) [inline src/app/configurator/configurator.component.html:52]: The idea was to distinguish the different form with the FormlyFieldConfig. Please look up at how that works.

File: src/app/configurator/configurator.component.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  width: 1px;
+  background-color: $line-color;
+  height: 100%;
+  margin: 0 auto;
+}
+.mat-dialog-actions {
+  margin-bottom: 0;
+}
+.leftt {
+  justify-content: flex-start !important;
```
Comment (knicken) [inline src/app/configurator/configurator.component.scss:16]: Please don't use !important. Also there's a type in the class name

File: src/app/configurator/configurator.component.ts [hunk 3]
```diff
@@ -1,20 +1,73 @@
...
 
 @Component({
   selector: 'app-configurator',
   templateUrl: './configurator.component.html',
   styleUrls: ['./configurator.component.scss']
 })
+
 export class ConfiguratorComponent implements OnInit {
 
+  configurator: string | undefined;
```
Comment (knicken) [inline src/app/configurator/configurator.component.ts:17]: Seems like this is the action (cta). If so please rename the variable to make it recognisable

File: src/app/configurator/configurator.component.ts [hunk 4]
```diff
@@ -1,20 +1,73 @@
...
+    this.dialogRef.close();
+  }
+
+  submit() {
+    if (this.configuratorForm.valid) {
+      this.configuratorService.addToCart({
+        ...this.configuratorForm.value,
+        productId: this.productId,
+        cta: this.configurator,
+        country: 'example-country'
```
Comment (knicken) [inline src/app/configurator/configurator.component.ts:52]: This should be the selected country

**Review by knicken — CHANGES_REQUESTED** [review]
Theres 3 minor issues and the FormlyFieldConfig which is quite important to understand for the future.

File: src/app/configurator/configurator.component.scss (hunk 2 above)
Comment (NSakuta) [reply src/app/configurator/configurator.component.scss:16]: sorry, I know. I don't use this class. I removed it

File: src/app/configurator/configurator.component.ts (hunk 3 above)
Comment (NSakuta) [reply src/app/configurator/configurator.component.ts:17]: done

File: src/app/configurator/configurator.component.ts (hunk 4 above)
Comment (NSakuta) [reply src/app/configurator/configurator.component.ts:52]: done

File: src/app/configurator/configurator.component.html (hunk 1 above)
Comment (NSakuta) [reply src/app/configurator/configurator.component.html:52]: unfortunately this is not done :)

**Review by knicken — APPROVED** [review]

**Review by knicken — APPROVED** [review]

**Pull request merged**